flink-libraries/flink-table/src/test/scala/org/apache/flink/table/runtime/batch/sql/SetOperatorsITCase.scala

```diff
@@ -278,7 +278,14 @@ class SetOperatorsITCase(
     TestBaseUtils.compareResultAsText(results.asJava, expected)
   }
 
+  /**
+    * This test will checks IN for NULLs based on whether COUNT (*) and COUNT (a) are equal. Due to
+    * [[org.apache.flink.table.plan.rules.dataSet.DataSetAggregateWithNullValuesRule]] will
+    * union a NULL row in to input DataSet for non-groupBy agg. That caused COUNT (*) and COUNT(a)
+    * are not equal. So this test case ignored before FLINK-8355 be fixed.
```

Comment:
I think this test case is bug of batch, so we should open it after FLINK-8355. 
You are right we not sure how to fix the FLINK-8355 current time, but we are sure add a `null` row is not correct(best) way to deal with `non-groupby` query.  But I think we should fix FLINK-8355 after this PR merged(when we know how to fix it). What do you think?

Reply:
A workaround way is we create a `transformToBatchAggregateFunctions` or add a parameter to `transformToAggregateFunctions` which can identify batch or stream.
but not elegant :(.

Reply:
I had a look at the test and think that it is correct.  `SELECT a FROM Table3` returns `1, 2, 3` and `SELECT d FROM Table5` returns `1, 2, 2, 3, 3, 3, 4, 4, 4, 4, 5, 5, 5, 5, 5`. So the result of the query should be 6 times `true` and 9 times `false`.

Which result woud you expect?

Reply:
`6` times true and `9` times false is my expect.. but because `DataSetAggregateWithNullValuesRule` we will get the `9` times `null`.  as follows:
```
 expected: [false, false, false, false, false, false, false, false, false, true, true, true, true, true, true]
 received: [null, null, null, null, null, null, null, null, null, true, true, true, true, true, true] 
```

Reply:
The rule or anything else that happens during optimization does not matter. The only thing that counts are input data, query, and output data. Given the query and the input data, the current result is correct. 

The test was passing before. If it fails now, it is clear that the PR broke it.

Reply:
The test was passing before because both `COUNT(*)` and `COUNT(a)` ignore the `NULL` before this PR. That's a bug.... please seee SQL 92 https://docs.microsoft.com/en-us/sql/t-sql/functions/count-transact-sql 
```
COUNT(*) returns the number of rows in a specified table without getting rid of duplicates. It counts each row separately. This includes rows that contain null values.
```

Reply:
It is correct, that the current state does not work for all queries and input data. There is a bug that needs to be fixed. However, it is working correctly for the *given* test.

The test is correct and is now failing due to this PR. Hence it causes a regression. Your changes fix a bug but introduce another one.

Reply:
Maybe I did not clearly express what I mean. :)

Reply:
No worries. I think if we fix FLINK-8355 first, the test cases should work correctly. I've added a comment to FLINK-8355 with an idea how to fix it.